t/api/04-jobs.t

```diff
@@ -29,13 +29,15 @@ use OpenQA::File;
 use OpenQA::Parser 'parser';
 use OpenQA::Test::Case;
 use OpenQA::Jobs::Constants;
+use OpenQA::JobDependencies::Constants;
 use OpenQA::Log 'log_debug';
 use OpenQA::Script::CloneJob;
 use Mojo::IOLoop;
 use Mojo::File qw(path tempfile tempdir);
 use Digest::MD5;
 
-OpenQA::Test::Case->new->init_data(fixtures_glob => '01-jobs.pl 02-workers.pl 03-users.pl 05-job_modules.pl');
+OpenQA::Test::Case->new->init_data(
+    fixtures_glob => '01-jobs.pl 02-workers.pl 03-users.pl 05-job_modules.pl 06-job_dependencies.pl');
```

Comment:
If we could live without the job dependencies in this test module it might be a better idea to keep it this way.

For the sake of completeness, the following test modules use `06-job_dependencies.pl` already:

* t/10-jobs.t
* t/37-limit_assets.t
* t/ui/16-tests_dependencies.t

so not many but also no API test that we should reuse.

I would assume a dedicated test module that *only* includes `06-job_dependencies.pl` or at best no fixtures at all and we just create what we need for the test would be fast and efficient. Also you are updating the job from the fixture already so IMHO better to explicitly create just what is needed in the right way. This could help us to reduce inter-dependencies between tests and fixtures.

Reply:
> Also you are updating the job from the fixture already so IMHO better to explicitly create just what is needed in the right way.

Creating the needed dependencies explicitly seems more reasonable, indeed. I've only come up this way because I'm still used to have all fixtures everywhere and only discovered afterwards that some are missing.